### PR TITLE
Fix:ecommerce lookml

### PR DESCRIPTION
**Pull request by axelazaid**
* update `.lkml` files to include measures and dimensions in the ecommerce package
  -  added descriptions to dimensions and measures

### CODE REVIEW

**Comment by jthandy** [comment]
@axelazaid I actually have essentially no background with this package so I'll let @eogilvy12 do the review here!

File: lookml/customers.view.lkml [hunk 1]
```diff
@@ -1,37 +1,24 @@
...
-  dimension: phone {
-    type: string
-    sql: ${TABLE}.phone ;;
-  }
-
-# ------------------------------------------------ Dates
+#------------------------------------------ DATES
 
-  dimension_group: created {
+  dimension_group: created_date {
```
Comment (eogilvy12) [inline lookml/customers.view.lkml]: I think you should have this field be `created` to keep consistent with removing `date` from your timestamps

File: lookml/customers.view.lkml [hunk 2]
```diff
@@ -59,4 +47,227 @@ view: customers {
...
+      month,
+      quarter,
+      year
+    ]
+    sql: ${TABLE}.updated_at ;;
+  }
+
+#------------------------------------------ CUSTOMER ATTRIBUTES
+
+  dimension: first_name {
```
Comment (eogilvy12) [inline lookml/customers.view.lkml]: remove first name, last name and phone

File: lookml/ecommerce.model.lkml [hunk 3]
```diff
@@ -3,25 +3,24 @@ connection: " "
 include: "*.view.lkml"         # include all views in this project
 include: "*.dashboard.lookml"  # include all dashboards in this project
 
+#------------------------------------------ ECOMMERCE
+
+explore: customers{
+  label: "Shopify Customers"
```
Comment (eogilvy12) [inline lookml/ecommerce.model.lkml]: remove label. this is for the ecom package and will not always be shopify data!

File: lookml/ecommerce.model.lkml [hunk 4]
```diff
@@ -3,25 +3,24 @@ connection: " "
...
+}
+
 explore: orders {
+  label: "Shopify Orders"
```
Comment (eogilvy12) [inline lookml/ecommerce.model.lkml]: remove shopify

File: lookml/order_items.view.lkml [hunk 5]
```diff
@@ -1,58 +1,67 @@
...
   }
-# ------------------------------------------------ Foreign Keys
+
+#------------------------------------------ IDS
+
   dimension: order_id {
-    type: number
-    hidden: yes
+    description: "This is the order ID"
+    type: string
```
Comment (eogilvy12) [inline lookml/order_items.view.lkml]: i think this should be a number

File: lookml/order_items.view.lkml [hunk 6]
```diff
@@ -1,58 +1,67 @@
...
     sql: ${TABLE}.order_id ;;
   }
-
+  
   dimension: product_id {
-    type: number
     hidden: yes
+    type: string
```
Comment (eogilvy12) [inline lookml/order_items.view.lkml]: i think this should also be a number normally.

File: lookml/order_items.view.lkml [hunk 7]
```diff
@@ -1,58 +1,67 @@
...
-  dimension: sku {
-    type: string
-    sql: ${TABLE}.sku ;;
-  }
+#------------------------------------------ ORDER ATTRIBUTES
 
-  dimension: quantity {
-    type: number
-    sql: ${TABLE}.quantity ;;
+  dimension: product_sku {
```
Comment (eogilvy12) [inline lookml/order_items.view.lkml]: this is more frequently called sku not product_sku

File: lookml/orders.view.lkml [hunk 8]
```diff
@@ -1,36 +1,42 @@
...
-    hidden: yes
-    sql: ${TABLE}.billing_address_id ;;
-  }
+#------------------------------------------ IDS
 
   dimension: customer_id {
-    type: number
-    hidden: yes
+    description: "This is the user's customer ID"
+    type: string
```
Comment (eogilvy12) [inline lookml/orders.view.lkml]: i think this is normally a number as well

File: lookml/orders.view.lkml [hunk 9]
```diff
@@ -1,36 +1,42 @@
...
     sql: ${TABLE}.customer_id ;;
   }
+  
+#------------------------------------------ DATES
 
-  dimension: shipping_address_id {
-    type: number
-    hidden: yes
-    sql: ${TABLE}.shipping_address_id ;;
+  dimension_group: created_at {
```
Comment (eogilvy12) [inline lookml/orders.view.lkml]: i think we may want to rename this created

File: lookml/orders.view.lkml [hunk 10]
```diff
@@ -70,51 +95,254 @@ view: orders {
...
+  }
+
+  dimension: completed_order_number {
     type: number
+    sql: ${TABLE}.completed_order_number ;;
+    group_label: "Order Attributes"
+  }
+
+  dimension: is_completed {
+    description: "This returns '1' if order is completed (financial_status in ('paid', 'partially_paid', 'partially_refunded')), '0' if no"
```
Comment (eogilvy12) [inline lookml/orders.view.lkml]: update this description- this is very specific to shopify. it should say this returns 1 if the parameters for a completed order have been fulfilled

File: lookml/orders.view.lkml [hunk 11]
```diff
@@ -70,51 +95,254 @@ view: orders {
...
+    type: number
+    sql: ${TABLE}.is_completed ;;
+    group_label: "Order Attributes"
+  }
+
+  dimension: order_number {
```
Comment (eogilvy12) [inline lookml/orders.view.lkml]: I think this is a shopify field. please check on this. order_seq_number is hidden above but it is what we create as the order number

File: lookml/orders.view.lkml [hunk 12]
```diff
@@ -70,51 +95,254 @@ view: orders {
...
     type: string
-    sql: ${TABLE}.new_or_repeat ;;
+    sql: ${TABLE}.new_vs_repeat ;;
+    group_label: "Customer Attributes"
   }
 
-  dimension: months_from_start {
+#------------------------------------------ DIMENSIONS
+
+  dimension: billing_address_id {
```
Comment (eogilvy12) [inline lookml/orders.view.lkml]: I think you should remove billing shipping and location id - these are shopify specific

File: lookml/products.view.lkml [hunk 13]
```diff
@@ -1,35 +1,85 @@
...
-    type: date
-    sql: ${TABLE}.deleted_at ;;
+  
+  dimension: vendor {
+    type: string
+    sql: ${TABLE}.vendor ;;
   }
 
-# ------------------------------------------------ Measures
+# ------------------------------------------------ MEASURES
```
Comment (eogilvy12) [inline lookml/products.view.lkml]: I don't think we need a measure on this table

File: lookml/products.view.lkml [hunk 14]
```diff
@@ -1,35 +1,85 @@
...
-  dimension: created_at {
-    type: date
-    sql: ${TABLE}.created_at ;;
-  }
-
-  dimension: deleted_at {
-    type: date
-    sql: ${TABLE}.deleted_at ;;
+  
+  dimension: vendor {
```
Comment (eogilvy12) [inline lookml/products.view.lkml]: vendor is specific to shopify

**Review by eogilvy12 — CHANGES_REQUESTED** [review]
@alexa I put comments within each file. let me know if you have questions!

once we have this updated, we need to update the dashboard lookml file and test that these changes are maintained before merging in

**Review by eogilvy12 — APPROVED** [review]
this looks excellent! now we can move onto the dashboard lookml updating. I think we should include that in this PR so do not merge this in yet - let's chat about how to best tackle the next step.

**Review by eogilvy12 — CHANGES_REQUESTED** [review]
Can you schedule 15 minutes to review the output of this with me? @axelazaid I think we should add a few reports to the main dashboard

**Comment by clrcrl** [comment]
Closing due to deprecation

**Pull request closed without merging**